src/components/Personalization/createApplyPropositions.js

```diff
@@ -12,61 +12,78 @@ governing permissions and limitations under the License.
-  const updatePropositions = ({ proposition, metadataForScope }) => {
-    proposition.items.forEach(item => {
-      if (item.schema.includes(HTML_CONTENT_ITEM)) {
-        item.data.selector =
-          metadataForScope.selector || DEFAULT_METADATA.selector;
-        item.data.type = metadataForScope.actionType || DEFAULT_METADATA.type;
-      }
-    });
+export default ({ executeDecisions }) => {
+  const filterItemsPredicate = item => {
+    if (SUPPORTED_SCHEMAS.includes(item.schema)) {
+      return true;
+    }
+    return false;
+  };
+
+  const updatePropositionItems = ({ proposition, metadataForScope }) => {
+    proposition.items = proposition.items
+      .filter(filterItemsPredicate)
+      .map(item => {
+        if (item.schema === HTML_CONTENT_ITEM && isObject(metadataForScope)) {
+          item.data.selector = metadataForScope.selector;
+          item.data.type = metadataForScope.actionType;
+        }
+        return item;
+      });
     return proposition;
   };
 
+  const filterPropositionsPredicate = proposition => {
+    if (proposition.scope === PAGE_WIDE_SCOPE && proposition.renderAttempted) {
+      return false;
+    }
+
+    return true;
+  };
+
   const preparePropositions = ({ propositions, metadata }) => {
-    return Promise.resolve(
-      propositions.map(proposition => {
-        const completeProposition = { ...proposition };
+    return propositions
+      .filter(filterPropositionsPredicate)
+      .map(proposition => {
+        const completeProposition = (({ id, items, scope, scopeDetails }) => ({
+          id,
+          items,
+          scope,
+          scopeDetails
+        }))(proposition);
         if (isNonEmptyArray(completeProposition.items)) {
-          const metadataForScope =
-            !isEmptyObject(metadata) &&
-            isObject(metadata[completeProposition.scope])
-              ? metadata[completeProposition.scope]
-              : DEFAULT_METADATA;
-          return updatePropositions({
+          return updatePropositionItems({
             proposition: completeProposition,
-            metadataForScope
+            metadataForScope: metadata[completeProposition.scope]
           });
         }
         return completeProposition;
       })
-    );
+      .filter(proposition => isNonEmptyArray(proposition.items));
   };
 
   const applyPropositions = ({ propositions, metadata }) => {
-    return preparePropositions({ propositions, metadata })
-      .then(completePropositions => executeDecisions(completePropositions))
-      .then(() => {
-        showContainers();
-        propositions.forEach(proposition => delete proposition.renderAttempted);
-        return composePersonalizationResultingObject(propositions, true);
-      });
+    const propositionsToExecute = preparePropositions({
+      propositions,
+      metadata
+    });
+    return executeDecisions(propositionsToExecute).then(() => {
+      return Promise.resolve(
```

Comment:
No need to wrap this in Promise.resolve. `executeDecisions` already returns a promise.

Reply:
@jonsnyder Tests fail with the error "applyPropositions(...).then is not a function" if I remove the Promise.resolve().  I didn't think I would need Promise.resolve() here either, but I think it's because we aren't using the promise value returned by executeDecisions.

Reply:
Ah, I see. This is because the mock object returned from executeDecisions in the test is not a real promise. i.e:

```
diff --git a/test/unit/specs/components/Personalization/createApplyPropositions.spec.js b/test/unit/specs/components/Personalization/createApplyPropositions.spec.js
index be6e5dc5..43b605d8 100644
--- a/test/unit/specs/components/Personalization/createApplyPropositions.spec.js
+++ b/test/unit/specs/components/Personalization/createApplyPropositions.spec.js
@@ -116,11 +116,7 @@ describe("Personalization::createApplyPropositions", () => {
   });
 
   it("it should return an empty propositions promise if propositions is empty array", () => {
-    const executeDecisionsPromise = {
-      then: callback => callback(PAGE_WIDE_SCOPE_DECISIONS)
-    };
-
-    executeDecisions.and.returnValue(executeDecisionsPromise);
+    executeDecisions.and.returnValue(Promise.resolve(PAGE_WIDE_SCOPE_DECISIONS));
 
     const applyPropositions = createApplyPropositions({
       executeDecisions
```